volatility3/framework/layers/qemu.py

```diff
@@ -217,9 +229,11 @@ def _decode_data(self, data: bytes, mapped_offset: int, offset: int, output_leng
         of the starting data.  It is the responsibility of the layer to turn the provided data chunk into the right
         portion of data necessary.
         """
-        start_offset = offset ^ (offset & 0xfff)
+        page_size = self._configuration.get('page_size', None) or 4096
```

Comment:
Is there ever a time this value could be 0?  I'm trying to figure out why (even in our original code), we set a default of None and then or it with 4096.  I'd probably just set the default of `4096`, but there's the very specific situation where `page_size` is 0 that this catches but a default of `4096` would not...

Reply:
I can't imagine a situation where `page_size` would be 0, since data is stored in chunks of `page_size` bytes. The only thing that could happen is that the value would be set to 0 due to errors while reading the configuration JSON, but that should (I guess) be handled in the function `_read_configuration`. Would you like me to change the `None` to `4096`?

Reply:
Errr, yeah, I think it would be easier to read, but it occurs to me we could just force `_read_configuration` to populate the `page_size` entry whether there was one or not.  We could also deal with errors there too?  If you'd be happy doing that, then I think it might be the best way of dealing with it (if you'd prefer me to make the changes, just shout).  Thanks!  5:)

Reply:
Issue is fixed now I think (after my force-pushed commit). An exception is raised if reading the configuration fails, and if no JSON is found at the end of the file (= no config is present), an emtpy dict is returned. I slightly prefer leaving the config as close to the original, rather than forcefully populating it. :relieved: